[ScorpiusSigma] ip

ScorpiusSigma opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ScorpiusSigma committed:
Level-1: Greet, Echo, Exit

ScorpiusSigma committed:
[Added] .class files to .gitignore

ScorpiusSigma committed:
[Added] .class files to .gitignore

ScorpiusSigma committed:
Level-2: Add, List

ScorpiusSigma committed:
Level-2: Add, List

ScorpiusSigma committed:
[Removed] lower case and extra indent

ScorpiusSigma committed:
Merge branch 'master' of https://github.com/ScorpiusSigma/ip

# Conflicts:
#	src/main/java/Duke.java

ScorpiusSigma committed:
Level-2: Add, List

ScorpiusSigma committed:
[Added] Echo to user input

ScorpiusSigma committed:
Level-3: Mark as Done

ScorpiusSigma committed:
[Added] \n after all print

ScorpiusSigma committed:
Level-4: ToDo, Event, Deadline

ScorpiusSigma committed:
A-TextUiTesting

ScorpiusSigma committed:
[Added] DukeException

ScorpiusSigma committed:
Level-5: Handle Errors

ScorpiusSigma committed:
Level-6: Delete

ScorpiusSigma committed:
restructure app

ScorpiusSigma committed:
Move bye() out of handleRequest() to  main()

ScorpiusSigma committed:
Add request class

ScorpiusSigma committed:
Move request to request package folder

ScorpiusSigma committed:
Request class: remove import CommandAPDU

ScorpiusSigma committed:
Edit typo from unwrwap() to unwrap()

ScorpiusSigma committed:
Add ToDoRequest class

ScorpiusSigma committed:
Request class: change values to value

ScorpiusSigma committed:
Add RequestException

ScorpiusSigma committed:
Request: change checkRequestRequirment return void

ScorpiusSigma committed:
Add EventRequest

ScorpiusSigma committed:
Add Deadline Request class

ScorpiusSigma committed:
Add Database class and DatabaseException class

ScorpiusSigma committed:
Refactor code base

ScorpiusSigma committed:
Duke class: Add print to functions

ScorpiusSigma committed:
Database class: Remove prints

ScorpiusSigma committed:
ToDoRequest class: Join returned description

ScorpiusSigma committed:
Duke class: Remove getUserInput function

ScorpiusSigma committed:
Database class: Add loadFromFile Method

ScorpiusSigma committed:
Database class: change method name, remove generic

ScorpiusSigma committed:
Task subclasses: add getter for attributes

ScorpiusSigma committed:
Database class: add loadFromFile and updateFile

ScorpiusSigma committed:
Database class: remove updateFile and add close()

close method will handle the updating of database when the program ends

ScorpiusSigma committed:
Duke class: add taskDB.close() to bye()

ScorpiusSigma committed:
Database class: move private methods to the bottom

ScorpiusSigma committed:
add DukeDateTimeFormatter class

ScorpiusSigma committed:
Deadline and Event Request class: format datetime

ScorpiusSigma committed:
Merge branch 'branch-Level-7'

ScorpiusSigma committed:
Storage class: rename Database to Storage

Storage class now handles the deserialisation, deserialisation, loading and unloading of data from file only

ScorpiusSigma committed:
Add Ui class to handle user I/O

ScorpiusSigma committed:
Add TaskList an inheritance of ArrayList<Task>

ScorpiusSigma committed:
Change DatabaseException to StorageException

ScorpiusSigma committed:
change RequestException to CommandException

ScorpiusSigma committed:
Change Request to Command

ScorpiusSigma committed:
Command class: add abstract method execute

ScorpiusSigma committed:
add subclasses of Commands

ScorpiusSigma committed:
Remove request package

ScorpiusSigma committed:
Add parser to make sense of the commands

ScorpiusSigma committed:
Refactor Duke to extract out closely related code as classes

ScorpiusSigma committed:
Edit input of text-ui-test/input.txt

ScorpiusSigma committed:
Created parser package

ScorpiusSigma committed:
Merge branch 'add-gradle-support'

ScorpiusSigma committed:
Duke class: remove loop label

ScorpiusSigma committed:
Setup Gradle

ScorpiusSigma committed:
Rename task package to tasks package

ScorpiusSigma committed:
Edit task package import to tasks package

ScorpiusSigma committed:
Add JUnit test for Deadline, Event and Todo

ScorpiusSigma committed:
Fix NullPointerException when calling execute

ScorpiusSigma committed:
Edit build.gradle to include mainClassName="Duke"

ScorpiusSigma committed:
Clean up by arranging the method orders

ScorpiusSigma committed:
Duke class: documentation

ScorpiusSigma committed:
Duke class: documentation added period

ScorpiusSigma committed:
Duke class: Fix indentation

ScorpiusSigma committed:
Ui class: Documentation

ScorpiusSigma committed:
Todo class: Documentation

ScorpiusSigma committed:
Duke class: remove @return on constructor doc

ScorpiusSigma committed:
Todo class: edit toString documentation

ScorpiusSigma committed:
Task class: documentation

ScorpiusSigma committed:
Event class: documentation

ScorpiusSigma committed:
Deadline class: documentation

ScorpiusSigma committed:
TaskList class: documentation

ScorpiusSigma committed:
Storage class: documentation

ScorpiusSigma committed:
Parser class: documentation

ScorpiusSigma committed:
Parser class: added s

ScorpiusSigma committed:
DateTimeParser class: documentation

ScorpiusSigma committed:
Documented DukeException and it's subclasses

ScorpiusSigma committed:
Document classes in command package

ScorpiusSigma committed:
Remove argument when instantiating ExitCommand

ScorpiusSigma committed:
Duke class: removed @return

ScorpiusSigma committed:
Edit to checkstyle standard except indentation

ScorpiusSigma committed:
Add find command

ScorpiusSigma committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/parser/Parser.java

ScorpiusSigma committed:
Change taskList.TaskList to tasklist.TaskList

ScorpiusSigma committed:
Merge branch 'branch-A-CodingStandard'

ScorpiusSigma committed:
Edit import path to duke.<path>

ScorpiusSigma committed:
Edit according to checkStyleMain checkStyleTest

zheng-ze commented on .vscode/settings.json:
```diff
@@ -0,0 +1,5 @@
+{
```
Should add this file into .gitignore file

ScorpiusSigma replied:
you're right, thanks!

zheng-ze commented on .prettierrc.json:
```diff
@@ -0,0 +1,7 @@
+{
```
This file too should be .gitignore.

ScorpiusSigma replied:
you're right, thanks!

zheng-ze commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,79 @@
+    */
+    public void handleRequest() {
+        while (true) {
+            try {
```
May I know why you are using try statement when you are not catching?

ScorpiusSigma replied:
Its an oversight. Initially, it was suppose to catch exception but then i changed the implementation. you're right, I should remove it. thanks!

ScorpiusSigma committed:
Edit based on the comments given by pzhengze

lukkesreysandeur commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,79 @@
+    private Ui ui;
+
+    /**
+    * Constructor.
```
Shouldn't there be a space between your comment header description and the parameter section? I think it applied for all comment headers.

lukkesreysandeur commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,138 @@
+    }
+
+    /**
+     * Retuns a boolean to tell if the command is an exist command.
```
Is this a typo? Looks like its supposed to be exit

lukkesreysandeur commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,42 @@
+            ui.showTaskRemoved(task, tasks.size());
+
+            storage.save(tasks);
+        } catch (CommandException | StorageException error) {
```
I like the use of space following the coding standard here.

lukkesreysandeur commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,107 @@
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.tasks.Event;
+import duke.tasks.Task;
```
Good separation and avoiding use of wildcard imports.

lukkesreysandeur commented on src/main/java/duke/command/UnMarkComamnd.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * UnMarkCommand class extends Command
+ */
+public class UnMarkComamnd extends Command {
```
Should you perhaps rename it to UnmarkCommand?

lukkesreysandeur commented on src/main/java/duke/parser/DateTimeParser.java:
```diff
@@ -0,0 +1,47 @@
+            LocalDateTime date = LocalDateTime.parse(datetimeString, formatFrom);
+            return date.format(DateTimeParser.formatTo);
+        } catch (DateTimeParseException e) {
+            System.out.println("Datetime has to be in the following format: yyyy-mm-dd HHmm");
```
Nice way to restrict the input to your command, I think i might do the same.

lukkesreysandeur commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,52 @@
+    public static Command parse(String input) {
+        Commands cmd = Command.getCommand(input);
+
+        switch (cmd) {
```
Good job following the switch statement convention.

lukkesreysandeur commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,52 @@
+        Commands cmd = Command.getCommand(input);
+
+        switch (cmd) {
+        case LIST:
```
Cool use of enums, I might consider using that too.

lukkesreysandeur commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,167 @@
+ */
+public class Storage {
+    private String cwd = System.getProperty("user.dir");
+    private java.nio.file.Path dataPath;
```
Perhaps you'd want to import this class as well?

lukkesreysandeur commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,167 @@
+     */
+    private String serialise(Task task) {
+        String data = "";
+        String isDone = (task.getIsDone() ? "1" : "0");
```
Maybe you want to rename this variable? Might be confusing as it is not actually a boolean.

lukkesreysandeur commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,167 @@
+        TaskList tasks = new TaskList();
+
+        try {
+            boolean directoryExists = java.nio.file.Files.exists(dataPath);
```
Perhaps this would be a clearer name?
```suggestion
            boolean doesDirectoryExist = java.nio.file.Files.exists(dataPath);
```

lukkesreysandeur commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,167 @@
+            // 3. Loops through the lines in the file.
+            while (scanner.hasNextLine()) {
+                String data = scanner.nextLine();
+                // 3a. deserialises the line.
```
I like this use of comments to explain the process step-by-step.

lukkesreysandeur reviewed (commented):
Overall followed the coding standard quite well. Minor nits here and there but otherwise I think that the coding conventions have been upheld. Also, did you accidentally commit your VScode extension files? I don't think that those config files should be commited.

ScorpiusSigma committed:
Setup JavaFX

ScorpiusSigma committed:
Add DialogBox

ScorpiusSigma committed:
Add view for dialog box and main window

ScorpiusSigma committed:
Edit command execute method to return String

ScorpiusSigma committed:
Add javaFX for Duke task bot

ScorpiusSigma committed:
Edit application width

ScorpiusSigma committed:
Edit Level 10 to make Jar extension work

ScorpiusSigma committed:
Command subclass: Add assertion in execute method

Command subclasses assumes parameter taken is correct and not null.

Having assertions will allow us to document important assumptions that should hold true which
can be used for bug tracking and validity checking.

ScorpiusSigma committed:
Merge pull request #1 from ScorpiusSigma/branch-A-Assertions

Command subclass: Add assertion in execute method

ScorpiusSigma committed:
Edit code according to comments on github

ScorpiusSigma committed:
Storage class: remove redundant comment

Storage class: remove redundant comment

Storage class has comments in load and save methods that describe codes that are already self-explanatory.

ScorpiusSigma committed:
Merge pull request #2 from ScorpiusSigma/branch-A-CodeQuality

Branch a code quality

ScorpiusSigma committed:
FindCommand class: Add streaming for filtering

FindCommand class: Add streaming for filtering

FindCommand uses for loop to find task that are contained in the list.

Loop is being replaced by streams

ScorpiusSigma committed:
Merge pull request #3 from ScorpiusSigma/branch-A-Streams

FindCommand class: Add streaming for filtering

ScorpiusSigma committed:
Add github workflows

ScorpiusSigma committed:
Command: Remove trailing spaces under assertion

ScorpiusSigma committed:
Change file name from UnMarkCommand to UnmarkCommand

ScorpiusSigma committed:
Task class: add updateDescription method

ScorpiusSigma committed:
DateTimeParser class: edit parse to throw error

ScorpiusSigma committed:
DateTimeParser class: add method to display error

ScorpiusSigma committed:
Edit Command to handle DateTimeParseException

ScorpiusSigma committed:
Add EditDescriptionCommand

ScorpiusSigma committed:
Add Edit Command type

ScorpiusSigma committed:
Remove datetime edit

ScorpiusSigma committed:
Merge pull request #4 from ScorpiusSigma/branch-C-Update

Branch c update

ScorpiusSigma committed:
Edit comments according to coding standards

ScorpiusSigma committed:
Merge branch 'branch-A-CodeQuality'

ScorpiusSigma committed:
DateTimeParser class: add throws to comment

ScorpiusSigma committed:
Change all duke reference to henz

ScorpiusSigma committed:
Change all duke references to henz

ScorpiusSigma committed:
Edit comments for Exception from duke to hez

ScorpiusSigma committed:
Edit DialogBox prefHeight to USE_COMPUTED_SIZE

ScorpiusSigma committed:
Add Ui.png

ScorpiusSigma committed:
Update docs/README.md

ScorpiusSigma committed:
DialogBox.fxml: remove USE_COMPUTED_SIZE

ScorpiusSigma committed:
ExitCommand class: exits UI when executed

ScorpiusSigma committed:
Update henz.txt

ScorpiusSigma committed:
Github workflow: remove windows and linux

ScorpiusSigma committed:
Github workflows: edit path

ScorpiusSigma committed:
Remove text in text-ui-test

ScorpiusSigma committed:
README.md: center image

ScorpiusSigma committed:
Remove test for text-ui-test

ScorpiusSigma committed:
Edit dialogbox.fxml  to not truncate text

ScorpiusSigma committed:
ExitCommand: save on execute

ScorpiusSigma committed:
Add break in README for readability
